CMakeLists.txt

```diff
@@ -62,6 +70,11 @@ set(CMAKE_CXX_STANDARD 20)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF)
 
+
+# Set global compiler and linker flags
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -stdlib=libc++")
+set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -stdlib=libc++")
```

Comment:
Not blocking this PR, just a note.
@tt-rkim @vtangTT @TT-billteng , I think ideally these should go away into a `toolchain.cmake` file as it depends on the toolchain (gcc / clang)

Reply:
Actually @vtangTT should be putting in a change into `cmake/macros.cmake` for this. We'll probably logic to use `stdlib=libstdc++` too for GCC

So this will change

Reply:
great!

Reply:
Logic has been added to `cmake/macros.cmake` switch stdlib for gcc vs clang.